app/src/main/java/com/example/fitappa/Exercise/Exercise/ExerciseTemplate.java

```diff
@@ -0,0 +1,77 @@
+package com.example.fitappa.Exercise.Exercise;
+
+import com.example.fitappa.Exercise.Set.RepSet;
+import com.example.fitappa.Exercise.Set.TimedSet;
+import com.example.fitappa.Exercise.Set.WeightedSet;
+
+import java.io.Serializable;
+
+public class ExerciseTemplate implements Serializable {
+    protected String name;
+    protected int numSets;
+    /* These values must be constant later
+    * Accepted values: REP, WEIGHTED, TIMED
+    * */
+    protected String category;
+
+    /**
+     * Constructor for a User class, takes in all necessary variables needed to make a User
+     *
+     * @param name   The String name referring to the name of the exercise
+     * @param sets   The int represents the number of sets
+     */
+    public ExerciseTemplate(String name, int sets, String category) {
+        this.name = name;
+        this.numSets = sets;
+        this.category = category;
+    }
+
+    // empty constructor necessary for firebase
+    @SuppressWarnings("unused")
+    public ExerciseTemplate() {
+    }
+
+    /**
+     * Creates an Exercise object given the name.
+     *
+     * @param name - name of the exercise
+     */
+    public ExerciseTemplate(String name) {
+        this.name = name;
+        this.numSets = 0;
+    }
+
+    /**
+     * Factory Design Pattern for Exercise objects.
+     * @return the appropriate Exercise object based on this.category
+     */
+    public Exercise<?> create() {
+        if (this.category.equals("REP"))
```

Comment:
maybe make this a switch statement for readability? I think lindsey recommended that in one lecture

Reply:
I had switch before but removed it because 
1. Longer code
2. I don't see the project ever exceeding more than 3 sets